src/lib/integrations/spotify/spotifyApi.ts

```diff
@@ -0,0 +1,154 @@
+import { SpotifyAuthService } from './spotifyAuth'
+
+interface SpotifyUserProfile {
+  email: string
+  display_name: string | null
+  id: string
+  product: string
+}
+
+interface SpotifyPlaylist {
+  id: string
+  name: string
+}
+
+const SPOTIFY_API_BASE = 'https://api.spotify.com/v1'
+
+export interface PlaybackState {
+  track_window: {
+    current_track: {
+      name: string;
+      artists: Array<{ name: string }>;
+    };
+  };
+  paused: boolean;
+  duration: number;
+  position: number;
+}
+
+export class SpotifyApiService {
+
+  private static spotifyApiRequest = async (path: string, options: RequestInit = {})=>{
```

Comment:
Made a function that makes the authenticated call to spotify. The other functions provide a path and options and the function handles the headers and response parsing